src/app.rs

```diff
@@ -149,6 +152,25 @@ impl Application {
                 let mut config =
                     config::load_from_paths(&config_paths, false).map_err(handle_config_errors)?;
 
+                // Conditionally wire up the 'tap' controller, based on API availability.
+                let tap_controller = if cfg!(feature = "api") {
+                    let tap_controller = TapController::new();
+
+                    let tap_sinks = config
```

Comment:
Thinking the issue with transforms is here.  I tried this locally and seemed to do the trick:

```
let tap_sinks = config
    .sources
    .keys().chain(config.transforms.keys())
    .map(|name| {
        (format!("_tap_{}", &name), tap_controller.make_sink(name))
    })
    .collect::<IndexMap<String, SinkOuter>>();
  ```

Reply:
Yep that’s it! Nicely caught 🎉

Reply:
I don't believe the `tap` is surviving a config change (hot update).  Oh, yep, you said that..

> Handling reloads. There's currently no mechanism to re-register components that may have changed.

Think that would be a nice follow up or include it here.  I think that's definitely needed if you a trying to use this to debug your config.